Runtime/Scripts/BaseMeshSync.cs

```diff
@@ -1077,98 +1071,210 @@ private protected void UpdateMaterialAssetByDefault(MaterialData src, ModelImpor
+                
+                if (tex != null) {
+                    destMat.EnableKeyword(keyword);
+                    return true;
                 }
-                else if (propName == _MetallicGlossMap)
-                {
-                    destMat.EnableKeyword(_METALLICGLOSSMAP);
+            }
+
+            destMat.DisableKeyword(keyword);
+
+            return false;
+        }
+
+        private static Dictionary<int, int[]> synonymMap = new Dictionary<int, int[]> {
+            { MeshSyncConstants._Color, new[] { MeshSyncConstants._BaseColor } },
+            { MeshSyncConstants._MainTex, new[] { MeshSyncConstants._BaseMap, MeshSyncConstants._BaseColorMap } },
+            { MeshSyncConstants._Glossiness, new[] { MeshSyncConstants._Smoothness, MeshSyncConstants._GlossMapScale } },
+            { MeshSyncConstants._BumpMap, new[] { MeshSyncConstants._NormalMap } },
+            { MeshSyncConstants._EmissionMap, new[] { MeshSyncConstants._EmissiveColorMap } },
+            { MeshSyncConstants._ParallaxMap, new[] { MeshSyncConstants._HeightMap } },
+            { MeshSyncConstants._BumpScale, new[] { MeshSyncConstants._NormalScale } }
+        };
+
+        private void ApplyMaterialProperty(Material destMat, 
+            List<TextureHolder> textureHolders,
+            MaterialPropertyData prop,
+            int propNameID, 
+            Dictionary<int, MaterialPropertyData> materialProperties) {
+           
+            // Shaders use different names to refer to the same maps, this map contains those synonyms so we can apply them to every shader easily:
+            if (synonymMap.TryGetValue(propNameID, out var synonyms)) {
+                foreach (var synonym in synonyms) {
+                    ApplyMaterialProperty(destMat, textureHolders, prop, synonym, materialProperties);
                 }
-                else if (propName == _BumpMap)
-                {
-                    destMat.EnableKeyword(_NORMALMAP);
+            }
+            
+            MaterialPropertyData.Type propType = prop.type;
+            if (!destMat.HasProperty(propNameID))
+                return;
+            
+            // Enable alpha test if there is a color texture so alpha clipping works:
+            if (propNameID == MeshSyncConstants._BaseMap ||
+                propNameID == MeshSyncConstants._MainTex || 
+                propNameID == MeshSyncConstants._BaseColorMap) {
+                bool hasAlpha = HandleKeywords(destMat, textureHolders, prop, MeshSyncConstants._ALPHATEST_ON);
+                if (hasAlpha) {
+                    destMat.SetOverrideTag(MeshSyncConstants.RenderType,MeshSyncConstants.TransparentCutout);
+                }
+#if AT_USE_HDRP
+                destMat.SetFloat(MeshSyncConstants._AlphaCutoffEnable, hasAlpha ? 1 : 0);
+                if (hasAlpha) {
+                    destMat.renderQueue = (int)RenderQueue.AlphaTest + 25;
```

Comment:
What is this doing, and what does `+ 25` mean?

Reply:
This is the render order that the hdrp pipeline uses when the material is set to transparent. I'm just replicating what that does. I think it would make the code even more complicated if we make a constant of every value we set on these shaders that's used in a single place.

Reply:
In this case, I think using literal number values is fine as long as it's clear in which render queue the process should be executed, which in this example is `AlphaTest`.

Reply:
Unless we have multiple rendering processes in the same `AlphaTest` category. I guess we don't, do we ?

Reply:
We don't have any need to set render orders for anything else. 2475 is what HDRP uses by default when the material is set to transparent, I wanted to make ours behave the same way.